optuna_dashboard/artifact/_backend.py

```diff
@@ -144,17 +144,50 @@ def upload_artifact_api(study_id: int, trial_id: int) -> dict[str, Any]:
             "artifacts": list_trial_artifacts(storage.get_study_system_attrs(study_id), trial),
         }
 
+    @app.post("/api/artifacts/<study_id:int>")
+    @json_api_view
+    def upload_study_artifact_api(study_id: int) -> dict[str, Any]:
```

Comment:
Is it possible to use `optuna.upload_artifact` in this function?

Reply:
Yes, it could. However, let's maintain the current changes for the moment. As @c-bata suggested, `upload_trial_artifact_api` could also be refactored to use `optuna.upload_artifact`. We should address these refinements in a follow-up PR.

Reply:
FYI, the `optuna.upload_artifact` function accepts `Study`, `Trial`, or `FrozenTrial` objects, but it only utilizes `study_id` and `trial_id`. In the current implementation, we retrieve the study object from the storage object and then extract the `study_id` from it. This approach is somewhat redundant and inefficient.

Reply:
> We should address these refinements in a follow-up PR.

Can you create a follow-up PR or issue not to forget this discussion? It is ok after this PR is merged.

Reply:
I created the issue #693.